design-library/src/components/BccProgress/BccProgress.vue

```diff
@@ -0,0 +1,74 @@
+<script lang="ts">
+export default {
+  inheritAttrs: false,
+};
+</script>
+
+<script setup lang="ts">
+import { computed, type StyleValue } from "vue";
+import BccFormLabel from "@/components/BccFormLabel/BccFormLabel.vue";
+
+type Props = {
+  value?: number;
+  max?: number;
+  size?: "sm" | "base" | "lg";
+  showValues?: boolean;
+  showPercentage?: boolean;
+  exceed?: boolean;
+};
+
+const props = withDefaults(defineProps<Props>(), {
+  value: 0,
+  max: 100,
+  size: "base",
+  showValues: true,
+  showPercentage: true,
+  exceed: true,
+});
+
+const percentage = computed(() => {
+  if (props.value > props.max && !props.exceed) return 100;
+  return (props.value / props.max) * 100;
+});
+
+const percentageSafe = computed(() => {
+  if (props.value > props.max) return 100;
+  return percentage.value;
+});
+
+const height = computed(() => {
+  return {
+    "h-1": props.size === "sm",
+    "h-1.5": props.size === "base",
+    "h-2": props.size === "lg",
```

Comment:
These can be custom classes as well, `bcc-progress-sm` etc. (just to make it easier for people using the library as a CSS library).